app-typescript/components/Button.tsx

```diff
@@ -0,0 +1,47 @@
+import * as React from 'react';
+import classNames from 'classnames';
+
+interface IProps {
```

Comment:
Implement icon-only buttons and standard text buttons separately according to these interfaces:

```
interface IButtonBase {
    theme: 'light' | 'dark';
    icon?: 'success' | 'alert'; // <-- add available icon names
    disabled?: boolean; // defaults to false
    type?: 'default' | 'primary' | 'success' | 'warning' | 'alert' | 'highlight' | 'sd-green';
    size?: 'small' | 'normal' | 'large'; // defaults to normal
    tooltip?: string;
    children?: never;
}

interface IPropsButton extends IButtonBase {
    text: string;
    expand?: boolean; // defaults to false
    style?: 'filled' | 'hollow' | 'text-only'; // defaults to filled
}

interface IPropsIconButton extends IButtonBase {
    shape?: 'square' | 'round'; // defaults to square
}
```

This is for taking advantage of the compiler to only allow valid configuration options. For example, the interface in this file has many properties and it's not obvious which can't be used together like`textOnly=true iconBtn=true`. Having separate components makes it possible to validate component configurations it at compile time.

Reply:
When I was referring to "icon only buttons" in the previous comment I meant "Buttons with icon font only"(btn--icon-only). Keep current `IconButton` as is, but create a separate component which would be for "Buttons with icon font only" and would use `IPropsIconButton` interface.

Reply:
In agreement with @pavlovicnemanja it's decides not to make separate component because it's going to be to much small components with only one props difference between them.

Reply:
You can reuse the same internal component. See this comment https://github.com/superdesk/superdesk-ui-framework/pull/389/files#r347453133

Reply:
Why? It is easier just to define `shape`... We don't want milion small components in UI-Framework. We want easy to use components. Let's keep it as this for now. Buttons are buttons.

Reply:
> Why?

Because `shape` doesn't work with text buttons by design. It's easier to understand and use the components when the interfaces are clean and all options work together. That's the main advantage against trying to put the components together by using CSS classes which might or might not work in different combinations. There might be cases when it's not possible to validate it with types only, but it's possible and quite easy in this case.

![screenshot_14](https://user-images.githubusercontent.com/2076953/69133525-56836a00-0ab6-11ea-8fe9-0c25430c54bb.png)

What's the issue with multiple components? It won't be a million, only a few more. Some of them won't even be exported like `ButtonInternal` example in the comment I linked.

Components will be used significantly more often than they will be modified, so it makes sense to prioritize ease of use for end-users(people using it in client-core or publisher) rather than developers of the framework who are already more familiar with how it works internally.